tf2_ros/src/buffer.cpp

```diff
@@ -164,8 +164,7 @@ Buffer::canTransform(
     (clock_->now() + rclcpp::Duration(3, 0) >= start_time) &&  // don't wait bag loop detected
     (rclcpp::ok()))  // Make sure we haven't been stopped (won't work for pytf)
   {
-    // TODO(sloretz) sleep using clock_->sleep_for when implemented
-    std::this_thread::sleep_for(std::chrono::milliseconds(10));
+    clock_->sleep_for(rclcpp::Duration(0, 1e+7));
```

Comment:
To me this is not very readable as 10ms.

Looks like `rclcpp::Duration` has a `std::chrono` [constructor](https://github.com/ros2/rclcpp/blob/rolling/rclcpp/include/rclcpp/duration.hpp#L45-L51).  I'd suggest this to be more readable:

```suggestion
    clock_->sleep_for(std::chrono::milliseconds(10));
```

or if that doesn't work, maybe this:

```suggestion
    clock_->sleep_for(rclcpp::Duration(std::chrono::milliseconds(10)));
```